qiskit_braket_provider/providers/adapter.py

```diff
@@ -347,11 +347,12 @@ def convert_qiskit_to_braket_circuit(circuit: QuantumCircuit) -> Circuit:
             # TODO: change Probability result type for Sample for proper functioning # pylint:disable=fixme
             # Getting the index from the bit mapping
             quantum_circuit.add_result_type(
-                result_types.Probability(
+                result_types.Sample(
+                    observable=Observable,
```

Comment:
Class `Observable` represents a quantum observable, but which one? You have to pass the computational basis observable

Reply:
Hi @dakk, Thank you for your comment. In my first commit, I used `observable=Observable.Z()` for the Probability; however, it was not passing the `tox -elint` test. I was changing codes to pass the `tox -elint`, which is why I got rid of the `Z()` part.

Reply:
What was the error of the linter?

Reply:
@dakk qiskit_braket_provider/providers/adapter.py:351:31: E1101: Class 'Observable' has no 'Z' member (no-member)

Reply:
I wonder if we have to use `StandardObservable` class instead of just `Observable` class

Reply:
No, it's correct to use `Observable.Z()`; maybe your braket version is outdated. This is my env, and Observable.Z is working:

![image](https://github.com/qiskit-community/qiskit-braket-provider/assets/1060425/f6333350-dccc-46d7-a17f-3affa4fb20dd)

Reply:
Right, seems like I have the same version. For that specific error, I added the comment `# pylint:disable=fixme` next to the code, and I think it can fix it. 

But, now I am getting after running `tox -elint` 

```
lint run-test: commands[3] | mypy --install-types --non-interactive .
error: --install-types failed (no mypy cache directory)
```
I tried to follow this instruction on [mypy-install-types-mre](https://github.com/tk-woven/mypy-install-types-mre), but only produces the same error. I might have to spend more time with it.

Reply:
Maybe there is a problem with the linter that can't detect dynamically added members; observables are dynamically added to Observable class (from braket side).

There is a workaround:

```python
from braket.circuits import observables
observables.Z()
```

If the problem is the linter, this should work.

Reply:
Thanks a lot for your help here @dakk!

Reply:
> I wrote a comment here before, but I might have accidentally erased it. Thank you for your help @dakk. I found out I was still getting the same error even after updating with observables.Z(). I was able to get it to work after re-cloning my repository. I decided to just stick with the solution you provided though! I appreciate your help! :)

Oh, that's weird! It was a pleasure!